woocommerce/woocommerce
# [Settings] Clean up naming and typings

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR:
* Fixes duplicate naming for `pageSettings` v. `settingsPage` which is of type `SettingsPage`. 
* The types in the `global.d.ts` file were updated so IDE's can read them
* Update naming to reflect types. Instead of `pages: type of window.wcSettings.admin.settingsPages` its now `settingsData: SettingsData`



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build works, tests pass
2. Enable settings feature flag and activate gutenberg
3. Settings should work, reflecting the route, and load without error

<img width="675" alt="image" src="https://github.com/user-attachments/assets/9aed4dc3-d72f-4026-8de9-11f11de3c1ed">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Updates unreleased feature

</details>
